keras/applications/resnet_rs.py

```diff
@@ -0,0 +1,919 @@
+"""
+
+def Conv2DFixedPadding(filters, kernel_size, strides, name = None):
+    """Conv2D block with fixed padding."""
+    if name is None:
+        name = backend.get_uid("conv")
+
+    def apply(inputs):
+        if strides > 1:
+            inputs = fixed_padding(inputs, kernel_size)
+        return layers.Conv2D(
+            filters=filters,
+            kernel_size=kernel_size,
+            strides=strides,
+            padding="same" if strides == 1 else "valid",
+            use_bias=False,
+            kernel_initializer=CONV_KERNEL_INITIALIZER,
+            name=name,
+        )(inputs)
+    return apply
+
+def STEM(bn_momentum: float = 0.0, bn_epsilon: float =1e-5, activation: str ="relu", name = None):
+    """ResNet-D type STEM block."""
+    if name is None:
+        name = "stem" + str(backend.get_uid("stem"))
+
+    def apply(inputs):
+        bn_axis = 3 if backend.image_data_format() == "channels_last" else 1
+
+        # First stem block
+        x = Conv2DFixedPadding(
+            filters=32, kernel_size=3, strides=2, name= name + "_stem_conv_1"
+        )(inputs)
+        x = layers.BatchNormalization(
+            axis=bn_axis,
+            momentum=bn_momentum,
+            epsilon=bn_epsilon,
+            name=name + "_stem_batch_norm_1",
+        )(x)
+        x = layers.Activation(activation, name= name + "_stem_act_1")(x)
+
+        # Second stem block
+        x = Conv2DFixedPadding(
+            filters=32, kernel_size=3, strides=1, name= name + "_stem_conv_2"
+        )(x)
+        x = layers.BatchNormalization(
+            axis=bn_axis,
+            momentum=bn_momentum,
+            epsilon=bn_epsilon,
+            name=name + "_stem_batch_norm_2",
+        )(x)
+        x = layers.Activation(activation, name= name + "_stem_act_2")(x)
+
+        # Final Stem block:
+        x = Conv2DFixedPadding(
+            filters=64, kernel_size=3, strides=1, name= name + "_stem_conv_3"
+        )(x)
+        x = layers.BatchNormalization(
+            axis=bn_axis,
+            momentum=bn_momentum,
+            epsilon=bn_epsilon,
+            name=name + "_stem_batch_norm_3",
+        )(x)
+        x = layers.Activation(activation, name= name + "_stem_act_3")(x)
+
+        # Replace stem max pool:
+        x = Conv2DFixedPadding(
+            filters=64, kernel_size=3, strides=2, name= name + "_stem_conv_4"
+        )(x)
+        x = layers.BatchNormalization(
+            axis=bn_axis,
+            momentum=bn_momentum,
+            epsilon=bn_epsilon,
+            name=name + "_stem_batch_norm_4",
+        )(x)
+        x = layers.Activation(activation, name= name + "_stem_act_4")(x)
+        return x
+
+    return apply
+
+
+def SE(in_filters: int, se_ratio: float =0.25,  expand_ratio: int = 1,name = None):
+    """Squeeze and Excitation block."""
+    bn_axis = 3 if backend.image_data_format() == "channels_last" else 1
+    if name is None:
+        name = "se" + str(backend.get_uid("se"))
```

Comment:
Use a f-string and add a `_` before the counter

Reply:
Done